linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/h2/IngressIdentifier.scala

```diff
@@ -0,0 +1,68 @@
+package io.buoyant.linkerd.protocol.h2
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.twitter.finagle.Stack.Params
+import com.twitter.finagle._
+import com.twitter.finagle.buoyant.Dst
+import com.twitter.finagle.buoyant.h2.Request
+import com.twitter.finagle.param.Label
+import com.twitter.util.Future
+import io.buoyant.config.types.Port
+import io.buoyant.k8s._
+import io.buoyant.linkerd.IdentifierInitializer
+import io.buoyant.linkerd.protocol.H2IdentifierConfig
+import io.buoyant.router.RoutingFactory._
+
+class IngressIdentifier(
+  pfx: Path,
+  baseDtab: () => Dtab,
+  namespace: Option[String],
+  apiClient: Service[http.Request, http.Response]
+) extends Identifier[Request] {
+
+  private[this] val unidentified: RequestIdentification[Request] =
+    new UnidentifiedRequest(s"no ingress rule matches")
+
+  private[this] val ingressCache = new IngressCache(namespace, apiClient)
+
+  override def apply(req: Request): Future[RequestIdentification[Request]] = {
+    val hostHeader = req.headers.get("Host").lastOption
```

Comment:
I think this should use `com.twitter.finagle.buoyant.h2.Headers.Authority` as the header namer